test.py

```diff
@@ -0,0 +1,118 @@
+            iterations=iterations, extraIteration=extra_iteration
+        )
+        result2 = wholenodeDiracBenchmark(
+            iterations=iterations, extraIteration=extra_iteration
+        )
+        result3 = wholenodeDiracBenchmark(
+            iterations=iterations + 1, extraIteration=extra_iteration
+        )
+
+        assert result["geometric_mean"] >= 0 and result["geometric_mean"] < 100
+        assert result["arithmetic_mean"] >= 0 and result["arithmetic_mean"] < 100
+        assert result["median"] >= 0 and result["median"] < 100
+
+        for i in result["raw"]:
+            assert abs(i - result2["raw"][k]) <= result2["raw"][k] * 20 / 100
+            assert i >= 0
+            assert i < 100
+            k = k + 1
+
+        k = 0
+        for i in result2["raw"]:
+            assert abs(i - result3["raw"][k]) <= result3["raw"][k] * 20 / 100
+            assert i >= 0
+            assert i < 100
+            k = k + 1
+
+    elif copies == "jobslot":
+        result = jobslotDiracBenchmark(
+            iterations=iterations, extraIteration=extra_iteration
+        )
+        result2 = jobslotDiracBenchmark(
+            iterations=iterations, extraIteration=extra_iteration
+        )
+        result3 = jobslotDiracBenchmark(
+            iterations=iterations + 1, extraIteration=extra_iteration
+        )
+
+        assert result["geometric_mean"] >= 0 and result["geometric_mean"] < 100
+        assert result["arithmetic_mean"] >= 0 and result["arithmetic_mean"] < 100
+        assert result["median"] >= 0 and result["median"] < 100
+
+        for i in result["raw"]:
+            assert abs(i - result2["raw"][k]) <= result2["raw"][k] * 20 / 100
+            assert i >= 0
+            assert i < 100
+            k = k + 1
+
+        k = 0
+        for i in result2["raw"]:
+            assert abs(i - result3["raw"][k]) <= result3["raw"][k] * 20 / 100
+            assert i >= 0
+            assert i < 100
+            k = k + 1
+    else:
+        result = multipleDiracBenchmark(
+            copies=int(copies), iterations=iterations, extraIteration=extra_iteration
+        )
+        result2 = multipleDiracBenchmark(
+            copies=int(copies), iterations=iterations, extraIteration=extra_iteration
+        )
+        result3 = multipleDiracBenchmark(
+            copies=int(copies), iterations=iterations + 1, extraIteration=extra_iteration
+        )
+
+        assert result["geometric_mean"] >= 0 and result["geometric_mean"] < 100
+        assert result["arithmetic_mean"] >= 0 and result["arithmetic_mean"] < 100
+        assert result["median"] >= 0 and result["median"] < 100
+
+        for i in result["raw"]:
+            assert abs(i - result2["raw"][k]) <= result2["raw"][k] * 20 / 100
+            assert i >= 0
+            assert i < 100
+            k = k + 1
+
+        k = 0
+        for i in result2["raw"]:
+            assert abs(i - result3["raw"][k]) <= result3["raw"][k] * 20 / 100
```

Comment:
And also could you add a variable `threshold = 20 / 100` and replace `20 / 100` by `threshold` please?